src/converse-register.js

```diff
@@ -241,6 +246,24 @@
                     return false;
                 },
 
+                chooseProvider: function () {
+                    /* Method called when registration_domain is provided
+                     * which loads the registration form directly
+                     */
+                    var form_help = document.getElementsByClassName('form-help');
+                    var reg_req = tpl_registration_request({
+                        cancel: '',
+                        info_message: _converse.__('Requesting a registration form from the XMPP server')
+                    });
+                    //form_help[0].insertAdjacentHTML('afterend', reg_req); @jcbrand not working for some reason
```

Comment:
As far as the rendering is concerned, this should render a spinner and then a <p> tag. When using jQuery, it does exactly that, but the javascript insertAdjacentHTML counterpart renders the spinner text together, spinning the entire thing. More technically, on doing some inspect element, it shows that the javascript code renders the <p> element inside the span element. I think this is mainly because of parsing problem for the `/>` in the html string at the end of the `<span>`.